lib/modules/datasource/go/index.ts

```diff
@@ -59,8 +59,9 @@ export class GoDatasource extends Datasource {
       return null;
     }
 
-    // ignore v0.0.0- pseudo versions that are used Go Modules - look up default branch instead
-    const tag = value && !value.startsWith('v0.0.0-2') ? value : undefined;
+    // ignore vX.Y.Z-(0.)? pseudo versions that are used Go Modules - look up default branch instead
+    const tag =
+      value && !/^v\d+\.\d+\.\d+-0?\.?.*/.test(value) ? value : undefined;
```

Comment:
```suggestion
      value && !/^v\d+\.\d+\.\d+-.*/.test(value) ? value : undefined;
```
this is the same regex, so this is too wide.

it also matches `v1.2.3-beta.1` which probably shouldn't be matched here.

Reply:
see https://regex101.com/r/jaFaCr/1
it doesn't match `v1.2.3-beta.1`
only `v1.2.3-012302103` and `v1.2.3-0.01230`

see PR description for more details
`vX.Y.Z+1-0.DATED_UPDATE(\d)-DIGEST(\w)`

Reply:
once there's a dash after the version tag, it's considered a digest pin, and should be extracted as digest

Reply:
> see https://regex101.com/r/jaFaCr/1
> it doesn't match `v1.2.3-beta.1`
> only `v1.2.3-012302103` and `v1.2.3-0.01230`
> 
> see PR description for more details
> `vX.Y.Z+1-0.DATED_UPDATE(\d)-DIGEST(\w)`

that's a different regex

Reply:
> once there's a dash after the version tag, it's considered a digest pin, and should be extracted as digest

where are docs about that behavior?

Reply:
https://go.dev/ref/mod#pseudo-versions